### PR TITLE
docs-animations

### PR DESCRIPTION
**Pull request by pepegar**
~To be merged after #212~ ready to merge now.

This PR tackles #50 .  We've created a microsite based on sbt-microsites, and some animations based on reftree to explain how optimizations work on the compiler.

### CODE REVIEW

File: modules/site/src/main/resources/microsite/data/menu.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+options:
+  - title: Compilation
```
Comment (patofp) [inline modules/site/src/main/resources/microsite/data/menu.yml:2]: When done the configuration I think a document explaining the flags should be added.

**Review by patofp — APPROVED** [review]
GJ!

File: modules/site/src/main/docs/compilation.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+layout: docs
+title: Compilation
+permalink: docs/compilation
+---
+
+# Compilation
+
+The process of querying the underlying Spark datasest is dictated by SparQL algebra.  Bellman expects a three (or four) column dataset representing a graph, in which the first column represents the _subject_,the second one the _predicate_, and the third one the _object_. It's optional to add a fourth column representing the _graph_ to which the edge belongs.
+
+What we do in Bellman is, for each triple in the **Basic Graph Pattern**, query the dataframe, and then join them given their common variables, for example:
```
[the remark this answers is not in the document]
Comment (pepegar) [reply modules/site/src/main/docs/compilation.md:11]: It's explained afterwards, I think.  In any case we can add more information here afterwards

File: modules/site/src/main/resources/microsite/data/menu.yml (hunk 1 above)
Comment (pepegar) [reply modules/site/src/main/resources/microsite/data/menu.yml:2]: Sure, we can add that too!

**Pull request merged**